docs/kb/semgrep-supply-chain/scanning-projects-with-multiple-lock-files.md

```diff
@@ -0,0 +1,108 @@
+---
+tags:
+  - GitHub Actions
+  - Lock files
+description: Scanning projects with multiple lock files in GitHub Actions
+---
+
+# Scanning projects with multiple lock files in GitHub Actions
+
+Your project may have many modules, each with a dependency descriptor file; for example, `pom.xml` if Maven is used as a build dependency tool.
+This [project](https://github.com/finos/legend-engine) complies with the above description.
+In this case, Semgrep will run a successful supply chain scan of all the lock files, one per module, if they are correctly generated.
+
+## Generating lock files as the previous step
+In the case of using Maven, the command to use is:
+```
+mvn dependency:tree -DoutputFile=maven_dep_tree.txt
+```
+
+And this step must be executed before calling Semgrep. So, a typical Jenkins pipeline can look like this:
+```
+pipeline {
+  agent any
+    environment {
+      SEMGREP_APP_TOKEN = credentials('SEMGREP_APP_TOKEN')
+      SEMGREP_BRANCH="${GIT_BRANCH}"
+    }
+    stages {
+      stage ('Generate-LockFile') {
+        steps {
+            withMaven(maven: 'maven') {
+              sh "mvn dependency:tree -DoutputFile=maven_dep_tree.txt"
+            }
+        }
+      }
+      stage('Semgrep-Scan') {
+        steps {
+                script {
+                    if (env.GIT_BRANCH == 'master') {
+                      sh 'pip3 install semgrep'
+                      sh 'semgrep ci'
+                    }  
+                }
+        }
+      }
+    }
+}
+```
+In the case of using GitHub Actions as a CI platform, then using artifacts to share files between jobs is required.
```

Comment:
```suggestion
If you are using GitHub Actions as a CI platform, using artifacts to share files between jobs is required.
```
I'd also move this line down to the next section where you talk about the process - otherwise it's a bit odd to give a Jenkins example and then talk about GitHub Actions.